symCache.py

```diff
@@ -0,0 +1,111 @@
+import os
+import cPickle as pickle
+from symLogging import LogDebug
+from symUtil import mkdir_p
+from symUtil import GetSymbolFileName
+
+class Cache(object):
+  def __init__(self, options):
+    self.sOptions = options
+
+  def Update(self, oldMRU, newMRU, symbols):
+    maxSize = self.MAX_SIZE
+    oldMruSet = set(oldMRU[:maxSize])
+    newMruSet = set(newMRU[:maxSize])
+
+    inserted = newMruSet.difference(oldMruSet)
+    evicted = oldMruSet.difference(newMruSet)
+
+    LogDebug(
+      "Evicting {} and inserting {} entries in {}".format(
+      len(evicted),
+      len(inserted),
+      self.__class__))
+
+    self.Evict(evicted)
+    self.Insert(inserted, symbols)
+
+class MemoryCache(Cache):
+  def __init__(self, options):
+    super(MemoryCache, self).__init__(options)
+    self.sCache = {}
+    self.sCacheSize = 0
+    self.MAX_SIZE = options["maxCacheEntries"]
+
+  def Evict(self, libs):
+    for key in libs:
+      self.sCache.pop(key)
+
+  def Insert(self, libs, symbols):
+    for lib in libs:
+      self.sCache[lib] = symbols[lib]
+
+  def Get(self, lib):
+    return self.sCache[lib]
+
+  def LoadCacheEntries(self, MRU, diskCache):
+    for lib in MRU[:self.MAX_SIZE]:
+      self.sCache[lib] = diskCache.Get(lib)
+
+class DiskCache(Cache):
+  def __init__(self, options):
+    super(DiskCache, self).__init__(options)
+    self.sOptions = options
```

Comment:
nit: you don't really need to store all the options, you only use maxCacheFiles and cachePath